### ISSUE TITLE
duplicate examples in TwentyNewsgroupsClustering(.v2)

### ISSUE THREAD

**Issue by KennethEnevoldsen** [embeddings-benchmark/mteb#407]
```python
from mteb import TwentyNewsgroupsClustering

task = TwentyNewsgroupsClustering()

task.load_data()

t = list(zip(task.dataset["test"]["sentences"][9], task.dataset["test"]["labels"][9]))

len(t)  # 10909
len(set(t))  # 5260 (about half)
```

@Muennighoff, this seems like a bug, am I missing something?

**Comment by Muennighoff** [comment]
It does seem odd to me, e.g. `Eco-Freaks forcing Space Mining.` appears 47 times. The embedding would be the same each time so I'm not sure what the added value is.

@loicmagne I think you created the dataset, do you have thoughts on this? Assuming that https://www.kaggle.com/datasets/crawford/20-newsgroups/data does not have these dups, maybe it was due to sampling with replacement from each category? I think we're missing the creation script https://github.com/embeddings-benchmark/mteb/tree/main/scripts/data.

**Comment by loicmagne** [comment]
Yes this is an issue that had already been identified in a HF issue: https://huggingface.co/datasets/mteb/twentynewsgroups-clustering/discussions/2#64218e9801c62c1e41fdca39

The TwentyNewsgroupsClustering dataset was derived from the [sklearn dataset](https://scikit-learn.org/0.19/datasets/twenty_newsgroups.html) as part of the [se-benchmark](https://github.com/nreimers/se-benchmark/blob/master/seb/tasks/Clustering/TwentyNewsgroupsClustering.py) which was a precursor benchmark of MTEB made by @nreimers . When we created MTEB I copied the existing datasets and so the duplicates stayed. I think Nils no longer has the original scripts anymore so I don't know the exact reason but it's likely that sampling was done with replacement

If we were to either remove the duplicates or resample correctly the subsets, that would make the current results obsolete, so we'd have to re-evaluate again most models, but maybe that's something that will be done with MMTEB ? WDYT @Muennighoff @KennethEnevoldsen 

I can do the PR if we decide to fix it, it would also be a good opportunity to fix the early datasets

**Comment by KennethEnevoldsen** [comment]
@loicmagne I seems like from (#481 and #424) that we might end up outdating previous clustering results to speed it up. @Muennighoff I am unsure how to do that best such. We might just reimplement all the datasets in a new folder clustering_fast_and_fixed and then make the transition all at once with MMTEB?

**Comment by KennethEnevoldsen** [comment]
This is a more widespread issue. I will close this in favor of #835
